neutron/test/convert_test.cpp

```diff
@@ -144,6 +144,34 @@ TEST(Convert, Tof_to_DSpacing) {
             tof.labels()["component_info"]);
 }
 
+TEST(Convert, DSpacing_to_Tof) {
+  /* Assuming the Tof_to_DSpacing test is correct and passing we can test the
+   * inverse conversion by simply comparing a round trip conversion with the
+   * original data. */
+
+  const Dataset tof_original = makeTofDataForUnitConversion();
+  const auto dspacing = convert(tof_original, Dim::Tof, Dim::DSpacing);
+  const auto tof = convert(dspacing, Dim::DSpacing, Dim::Tof);
+
+  /* Test coordinates */
+  /* Slicing is needed as conversion introduces the dependance on Dim::Position
+   */
+  EXPECT_EQ(tof.coords()[Dim::Tof].slice({Dim::Position, 0}),
+            tof_original.coords()[Dim::Tof]);
```

Comment:
Shouldn't you test all slices? Or use `broadcast` to bring the reference to the same shape?